flexget/plugins/search/1337x.py

```diff
@@ -0,0 +1,165 @@
+    errors = False
+
+    def get(self, url, params):
+        """
+        Wrapper to allow refreshing the cookie if it is invalid for some reason
+        :param url:
+        :param params:
+        :return:
+        """
+        cookies = None
+
+        response = requests.get(url, params=params, cookies=cookies)
+
+        return response
+
+    # urlrewriter API
+    def url_rewritable(self, task, entry):
+        url = entry['url']
+        if url.endswith('.torrent'):
+            return False
+        if url.startswith('http://1337x.to/'):
+            return True
+        return False
+
+    def url_rewrite(self, task, entry):
+        """
+            Gets the download information for 1337x result
+        """
+
+        url = entry['url']
+
+        log.info('1337x rewriting download url: %s' % url)
+
+        try:
+            page = self.get(url, None)
+            log.debug('requesting: %s', page.url)
+        except RequestException as e:
+            log.error('1337x request failed: %s', e)
+
+        soup = get_soup(page.content)
+
+        # Get the infohash - not needed but the code is here *shrugs*
+        #infohash = str(soup.find("div", class_="infohash-box").contents)
+        #infohash = infohash[-(len(infohash)-infohash.find(':'))+2:]
+        #infohash = infohash[:infohash.find(' ')]
+
+        magnetURL = str(soup.find("a", id="magnetdl").get('href')).lower()
+        torrentURL = str(soup.find("a", id="torrentdl").get('href')).lower()
+
+        # Can't update the title to the full title name because it breaks entry.remove
+        #title = str(soup.title.string).replace("Download Torrent ","").replace("| 1337x","")
+
+        entry['url'] = torrentURL
+        entry['urls'] = []
+        entry['urls'].append(torrentURL)
+        entry['urls'].append(magnetURL)
+
+    @plugin.internet(log)
+    def search(self, task, entry, config):
+        """
+            Search for entries on 1337x
+        """
+
+        if not isinstance(config, dict):
+            config = {}
+
+        order_by = ""
+        sorted = ""
+        if (isinstance(config.get('order_by'), str)):
+            if (config['order_by'] != "leechers"):
+                order_by = "/{0}/desc" . format(config['order_by'])
+                sorted = "sort-"
+
+        entries = set()
+
+        for search_string in entry.get('search_strings', [entry['title']]):
+
+            query = "{0}search/{1}{2}/1/" . format(sorted,quote(search_string.encode('utf8')), order_by)
+            log.debug('Using search params: {0}; ordering by: {1}' . format(search_string, order_by or "default"))
+            try:
+                page = self.get(self.base_url + query, None)
+                log.debug('requesting: %s', page.url)
+            except RequestException as e:
+                log.error('1337x request failed: %s', e)
+                continue
+
+            soup = get_soup(page.content)
+            if (soup.find('div', attrs={'class': 'tab-detail'}) != None):
```

Comment:
The outer parenthesis is not necessary. It's also best to do `X is not None` instead of `X != None`.